Maple2.Server.Game/Manager/Items/EquipManager.cs

```diff
@@ -174,19 +174,19 @@ public class EquipManager {
                 return false;
             }
 
-            if (item.Metadata.Limit.TransferType == TransferType.BindOnEquip) {
+            if (item.Metadata.Limit.TransferType == TransferType.BindOnEquip || item.Metadata.Limit.TransferType == TransferType.BindOnLoot) {
                 session.Item.Bind(item);
             }
 
-            item.Group = ItemGroup.Default;
+            item.Group = ItemGroup.Badge;
             item.Slot = -1;
             session.Field?.Broadcast(EquipPacket.EquipBadge(session.Player, item));
 
             if (item.Badge.Type == BadgeType.PetSkin) {
                 session.Pet?.BadgeChanged(item.Badge);
             }
 
-            return true;
+            return Badge.TryAdd(item.Badge.Type, item);
```

Comment:
Rather than making this call which could fail, could you follow the pattern used for gear equipping:
```cs
Badge[item.Badge.Type] = item;
```